### PR TITLE
Update Pintk random models operation

### PR DESCRIPTION
**Pull request by champagne-cmd**
This PR replaces #1118 and uses @dlakaplan's new `calculate_random_models` function, as well as new fake toas machinery. Time residuals are created in `calculate_random_models` based on the user inputting `return_time=True` to the function. The mean residual of each set of random residuals within the span of real data is then calculated within `pulsar.py` and subtracted out of each respective residual set before plotting.

### CODE REVIEW

File: src/pint/pintk/plk.py [hunk 1]
```diff
@@ -29,19 +29,18 @@
...
-#    "This should also show up. test click revert, turn params on and off, and prefit model"
-# )
+log.debug(
+    "This should also show up. test click revert, turn params on and off, and prefit model"
+)
 
 plotlabels = {
     "pre-fit": [
-        "Pre-fit residual",
+        r"Pre-fit residual ($\mu$s)",
```
Comment (scottransom) [inline src/pint/pintk/plk.py:38]: Does this mean that residuals are always in us now?

**Review by scottransom — COMMENTED** [review]
All looks pretty good.  Two things I noticed while playing around (which if super easy, can just be fixed with this, otherwise I'll make a new issue) is that trying to make the x-axis "years" caused an error.  And also, using the menu to change the DEBUG/INFO/WARNING level didn't do anything.

File: src/pint/pintk/plk.py (hunk 1 above)
Comment (champagne-cmd) [reply src/pint/pintk/plk.py]: Sorry - accidentally copied over old code! This should be fixed now - the y axis scales to us, ms, and s.

**Comment by champagne-cmd** [comment]
I'm not getting any error when I switch over to years on the x axis, either with random models or without - maybe it was sue to the outdated code I had accidentally copied over?

As for the log level not working, I do not know why that is now failing. Everything looks right to me, but I had originally implemented that with the Astropy logger, perhaps the built-in logger has some extra steps needed?

**Comment by paulray** [comment]
I just pulled this PR and I confirm that the log level selection does not seem to be working, but the years X-axis did not seem to cause any problem.

**Comment by paulray** [comment]
I fiddled for a while and read some about the python logging facility, but I couldn't make it work.  I suggest we handle that as a separate PR.

**Comment by scottransom** [comment]
Yeah, I can't reproduce the y-axis "year" thing now.  And I agree that the logging should go to another PR.  So I think this looks all good.

**Pull request merged**